### PR TITLE
causalimpact 0.2.6 :snowflake:

### PR DESCRIPTION
**Pull request by boldorider4**
causalimpact 0.2.6 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-6037]
- dev_url:        https://github.com/jamalsenouci/causalimpact//tree/v0.2.6
- conda_forge:    https://github.com/conda-forge/causalimpact-feedstock
- pypi:           https://pypi.org/project/causalimpact/0.2.6
- pypi inspector: https://inspector.pypi.io/project/causalimpact/0.2.6/

### Explanation of changes:

- new version number

[PKG-6037]: https://anaconda.atlassian.net/browse/PKG-6037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: What import failures do you see?

File: recipe/meta.yaml
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: You might need c too no?

Comment (boldorider4) [reply recipe/meta.yaml]: why is it necessary? in case some usage of pytensor involved the compilation of pure C?

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: Actually, we should remove this. It's an implementation detail of pytensor.

Comment (tombenes) [reply recipe/meta.yaml]: I agree with Jean, this should be added as runtime dep for pytensor

Comment (boldorider4) [reply recipe/meta.yaml]: so should I rebuild `pytensor` and hold off on this until it's rebuilt?

Comment (tombenes) [reply recipe/meta.yaml]: That'd be my understanding of this issue

File: recipe/meta.yaml
Comment (boldorider4) [reply recipe/meta.yaml]: I can attach the error.
Admittedly, it seems to occur only for `py311`. [This](https://discourse.pymc.io/t/importerror-cannot-import-name-gaussian-from-scipy-signal/14170) is where I found the hint that it's related to which combination of `arviz` and `scipy` used.

Comment (boldorider4) [reply recipe/meta.yaml]: [causalimpact-feedstock_osx-error_log.log](https://github.com/user-attachments/files/17991993/causalimpact-feedstock_osx-error_log.log)

File: recipe/meta.yaml
Comment (tombenes) [inline recipe/meta.yaml]:
```suggestion
```
This should not be necessary anymore?

Comment (boldorider4) [reply recipe/meta.yaml]: agreed.

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - numpy
```
As there is no `numpy` in `host`, and requirements say about a plain `numpy` https://github.com/jamalsenouci/causalimpact/blob/v0.2.6/requirements.in#L2

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - importlib-metadata  # [py<38]
```
see https://github.com/jamalsenouci/causalimpact/blob/v0.2.6/setup.cfg#L35

File: recipe/meta.yaml
Comment (tombenes) [inline recipe/meta.yaml]:
```suggestion
    - numpy {{ numpy }}  # [not (win and py==310)]
```
I think that's the syntax

Comment (boldorider4) [reply recipe/meta.yaml]: thanks!

File: recipe/meta.yaml
Comment (boldorider4) [reply recipe/meta.yaml]: unfortunately the build fails without the pinning from cbc.

**Comment by skupr-anaconda** [comment]
There is an unusual warning in the logs:
`WARNING (pytensor.configdefaults): g++ not detected!  PyTensor will be unable to compile C-implementations and will default to Python. Performance may be severely degraded. To remove this warning, set PyTensor flags cxx to an empty string.`

Does it mean that `pytensor` may have the wrong configurations?

**Comment by boldorider4** [comment]
> There is an unusual warning in the logs: `WARNING (pytensor.configdefaults): g++ not detected! PyTensor will be unable to compile C-implementations and will default to Python. Performance may be severely degraded. To remove this warning, set PyTensor flags cxx to an empty string.`
> 
> Does it mean that `pytensor` may have the wrong configurations?

Interesting that you caught that. If that's the case, I suppose it's either something wrong with `pytensor` or `causalimpact` needs to take care of its configuration. I would say the first case seems more likely.
However the test where C++ compilation is involved passes. It could also be the case that the G++ compiler is not found within the `causalimpact` package. I will have a look.

**Comment by tombenes** [comment]
> > There is an unusual warning in the logs: `WARNING (pytensor.configdefaults): g++ not detected! PyTensor will be unable to compile C-implementations and will default to Python. Performance may be severely degraded. To remove this warning, set PyTensor flags cxx to an empty string.`
> > Does it mean that `pytensor` may have the wrong configurations?
> 
> Interesting that you caught that. If that's the case, I suppose it's either something wrong with `pytensor` or `causalimpact` needs to take care of its configuration. I would say the first case seems more likely. However the test where C++ compilation is involved passes. It could also be the case that the G++ compiler is not found within the `causalimpact` package. I will have a look.

My understanding is things will work, but with degraded performance. I'm interested to see if this is how our previous `pytensor` was operating, as it didn't have g++ in deps.

EDIT:
On the other hand, without compiler in pytensor the test was failing, so I don't fully understand what's the difference between test failing and passing with error which says "g++ not detected".

**Review by skupr-anaconda — APPROVED** [review]

**Review by tombenes — APPROVED** [review]
LGTM assuming we want to ignore the warning

**Comment by boldorider4** [comment]
> > > There is an unusual warning in the logs: `WARNING (pytensor.configdefaults): g++ not detected! PyTensor will be unable to compile C-implementations and will default to Python. Performance may be severely degraded. To remove this warning, set PyTensor flags cxx to an empty string.`
> > > Does it mean that `pytensor` may have the wrong configurations?
> > 
> > 
> > Interesting that you caught that. If that's the case, I suppose it's either something wrong with `pytensor` or `causalimpact` needs to take care of its configuration. I would say the first case seems more likely. However the test where C++ compilation is involved passes. It could also be the case that the G++ compiler is not found within the `causalimpact` package. I will have a look.
> 
> My understanding is things will work, but with degraded performance. I'm interested to see if this is how our previous `pytensor` was operating, as it didn't have g++ in deps.
> 
> EDIT: On the other hand, without compiler in pytensor the test was failing, so I don't fully understand what's the difference between test failing and passing with error which says "g++ not detected".

Let me explain for posterity.

The compiler in run of `pytensor` is there to get `pytensor` to make use of `g++` at runtime, should its feature using a compiler be used.

The warning itself is about `pytensor` not being able to find `g++` at runtime with its basic configuration. I did a bit of digging and it seems that `pytensor` has a history of not detecting `g++` despite it being discoverable in the search path. See [this case](https://github.com/pymc-devs/pytensor/issues/234) for `win-64`. I did some debugging around it, and I can confirm the warning is being printed despite `g++` being discoverable. 

The good news is that as long as `pytensor.config.cxx` is set, `pytensor` will be able to use the compiler at runtime, which explains why those tests pass. That's also the case for `pytensor` on `linux-64` (I checked).

To me this is an ignorable warning.

**Pull request merged**
